tools/azsdk-cli/docs/cli-commands-guidelines.md

```diff
@@ -0,0 +1,416 @@
+- Log analysis and processing
+- Test analysis and reporting
+- Cleanup operations
+- Internal engineering workflows
+
+**Examples:**
+- `engsys cleanup artifacts`
+- `engsys codeowners validate`
+- `engsys logs analyze`
+- `engsys test-analysis report`
+
+### 2. **package** - Package Operations
+**Namespace:** `Azure.Sdk.Tools.Cli.Tools.Package`  
+**Command Group:** `SharedCommandGroups.Package`  
+**Verb:** `package`  
+
+For operations at the SDK package level. The package group has further sub-grouping for better organization:
+
+#### Core Package Operations:
+- Build/compile SDK code
+- Generate source code
+- Release packages
+- Update version information
+- Validate packages
+
+#### Sub-Groups:
+
+##### **readme** - README Operations  
+For generating and updating README files:
+- Generate README files
+- Update README content
+- Validate README format
+- README documentation
+
+##### **sample** - Sample Operations
+For generating and updating SDK samples:
+- Generate new samples
+- Update existing samples
+- Validate sample code
+- Sample documentation
+
+##### **test** - Test Operations
+For creating, updating, and running tests for SDK packages:
+- Create new tests
+- Update existing tests
+- Run tests for specific packages
+- Test validation and reporting
+
+**Examples:**
+- `package build --package-path ./sdk/storage`
+- `package check-release-ready --package-name azure-core --version 1.0.0`
+- `package generate --typespec-project ./specification/storage`
+- `package readme generate --package-path ./sdk/compute`
+- `package readme update --package-path ./sdk/keyvault --section getting-started`
+- `package release --package-name azure-core`
+- `package sample generate --package-path ./sdk/compute`
+- `package sample update --package-path ./sdk/storage --sample-name basic-usage`
+- `package show-details`
+- `package test generate --package-path ./sdk/storage --test-type <type>`
+- `package test run --package-path ./sdk/keyvault --test-suite integration`
+- `package validate --package-path ./sdk/keyvault`
+
+### 3. **github** - GitHub Operations
+**Namespace:** `Azure.Sdk.Tools.Cli.Tools.GitHub`  
+**Command Group:** Custom (no predefined group)  
+**Verb:** `github`
+
+For operations related to GitHub:
+- Create pull requests
+- Check PR status
+- Manage labels
+- Repository operations
+
+**Examples:**
+- `github create-pr --title "Update SDK" --description "Description"`
+- `github get-pr-details --pr 123`
+- `github labels sync`
+
+### 4. **release-plan** - Release Planning
+**Namespace:** `Azure.Sdk.Tools.Cli.Tools.ReleasePlan`  
+**Command Group:** Custom (no predefined group)  
+**Verb:** `release-plan` 
+
+For release planning and SDK coordination:
+- Create release plans
+- Update SDK details in release plans
+- Link SDK pull requests to release plans
+- Get release plan details
+- Generate SDK for all languages
+- Workflow management
+
+**Examples:**
+- `release-plan check-api-readiness --typespec-project ./spec/compute`
```

Comment:
what is this looking? if the typespec is approved? I would prefer to hide CLC concepts like api readiness